### PR TITLE
roachtest: add test with liquibase test harness

### LINKED ISSUE #56824: roachtest: add liquibase test harness to nightly
The test harness (https://github.com/liquibase/liquibase-test-harness) passes with CockroachDB. To run it, use `mvn test -DdbName=cockroachdb '-DdbVersion=20.2'`

### PR DESCRIPTION
**Pull request by rafiss**
This will make sure we don't regress in our liquibase support.

fixes https://github.com/cockroachdb/cockroach/issues/56824

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/liquibase.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import "context"
+
+// This test runs the Liquibase test harness against a single cockroach node.
+
```
Comment (RichardJCai) [inline pkg/cmd/roachtest/liquibase.go:16]: Extra blank line here

File: pkg/cmd/roachtest/liquibase.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		if c.isLocal() {
+			t.Fatal("cannot be run in local mode")
+		}
+		node := c.Node(1)
+		t.Status("setting up cockroach")
+		c.Put(ctx, cockroach, "./cockroach", c.All())
+		c.Start(ctx, t, c.All())
+
+		version, err := fetchCockroachVersion(ctx, c, node[0], nil)
+		if err != nil {
```
Comment (RichardJCai) [inline pkg/cmd/roachtest/liquibase.go:32]: nit: I've been using `require.NoError(t, err)` for these, do you think it's better to use these in the future?

**Review by RichardJCai — APPROVED** [review]
LGTM with nits

**Comment by rafiss** [comment]
tftr!

bors r=RichardJCai

**Pull request merged**